packages/gatsby-source-nacelle/src/utils/linkNestedContentReferences.js

```diff
@@ -1,9 +1,19 @@
+const isUnresolvableNacelleReference = require('./isUnresolvableNacelleReference');
+
 function findNestedReferences(contentObject = {}, parent = null, key) {
   if (Array.isArray(contentObject)) {
     // if it's an array & has a nacelleEntryId
     // it's got top level nodes that we can link to instead
     if (contentObject[0]?.nacelleEntryId) {
-      parent[`${key}___NODE`] = contentObject.map(
+      // filter out any unresolvable nacelle references since these won't exist in the Gatsby Node system and break builds.
+      let filteredContent = contentObject.filter((entry) => {
+        return !isUnresolvableNacelleReference(entry);
+      });
+      // if there are assets, don't try to link to prevent assets from being included in Node linking
+      if (filteredContent.some((entry) => entry.type === 'asset')) {
+        return;
+      }
```

Comment:
This is one part I'm not sure of. If there are mixed asset/content types, you _might_ get issues around types with this setup. But I think having both `parent.key` and `parent.key___NODE` causes issues with Gatsby.